### DISCUSSION TITLE
When a user account is deleted, how can I remove all records associated with that user?

### DISCUSSION THREAD

**Discussion by maxhis** [pocketbase/pocketbase#5219] [Q&A] [answered]
My app has a feature to delete user accounts. When a user chooses to delete their account, I need to simultaneously remove all data associated with them. 

For example, I have a database table named `posts`, where one of the fields is `author`, which is linked to the users table.

I tried the `onModelAfterDelete` and `onModelBeforeDelete` hooks:

```javascript
onModelBeforeDelete((e) => {
  $app.dao().db().delete("posts", $dbx.exp("author = {:author}", { author: e.model }));
}, "users");
```

and:

```javascript
onModelAfterDelete((e) => {
  $app.dao().db().newQuery("DELETE FROM posts WHERE author IS NULL").execute();
}, "users");
```

both not working.

I'm not sure if it is related to the API rule. Here is the current configuration. I also tried to leave the `delete rule` empty, without lucky either.

<img width="657" alt="2024-07-15 100215" src="https://github.com/user-attachments/assets/e5703d92-2a33-4e2c-8722-82e6323cfe54">

**Comment by dschissler** [comment]
Each collection has a cascade delete option that can be configured via a toggle in the web UI.  That can automatically do a lot for you.

**Comment by maxhis** [comment]
By reviewing the PocketBase database schema, I finally figured out the reason why the deletion wasn't successful: the default value for the relation field in PocketBase is `''`.

Here is the code that works correctly:

```js
onModelAfterDelete((e) => {
  $app.dao().db().newQuery("DELETE FROM posts WHERE author = ''").execute();
}, "users");
```

**Reply by ganigeorgiev** [reply, marked as answer]
@maxhis You don't have to do this (it also doesn't make sense as it will remove all posts that doesn't have an author associated to them). Raw SQL queries also don't fire realtime events. As @dschissler suggested you can make use of the "Cascade delete" `relation` field option and leave to PocketBase to delete the records for you:

![cascade_delete](https://i.imgur.com/mbdc5IO.png)

**Reply by maxhis** [reply]
Wow, I didn't notice the Cascade delete. It's much more elegant than my solution!
